### PR TITLE
feat: django 5 support for cms 3.11

### PR DESCRIPTION
**Pull request by protoroto**
## Description

This cherry-picks two commits from @fsbraun in develop-4 branch and fixes toolbar tests as Django 5.0 doesn't accepts anymore GET requests for logout view

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by protoroto** [comment]
@fsbraun I should've now fix all the build/tests errors, and cleaned up the code a bit

File: cms/forms/fields.py [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
 class LazyChoiceField(forms.ChoiceField):
-    def _set_choices(self, value):
-        # we overwrite this function so no list(value) is called
-        self._choices = self.widget.choices = value
 
-    choices = property(forms.ChoiceField._get_choices, _set_choices)
+    def __init__(self, *args, **kwargs):
+        warnings.warn("LazyChoiceField is deprecated. Use Django's ChoiceField instead.",
+                      DeprecationWarning, stacklevel=2)
+        super().__init__(*args, **kwargs)
```
Comment (fsbraun) [inline cms/forms/fields.py:31]:
```suggestion
```

Let us not deprecate LazyChoiceField in 3.11. yet.

File: test_requirements/django-5.0.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-r requirements_base.txt
+Django>=5.0a1,<5.1
```
Comment (fsbraun) [inline test_requirements/django-5.0.txt:2]:
```suggestion
Django>=5.0,<5.1
```
Django 5.0 is out now. :-)

**Review by fsbraun — CHANGES_REQUESTED** [review]
@protoroto Thanks so much for the PR! I left two small comments.

**Comment by protoroto** [comment]
> @protoroto Thanks so much for the PR! I left two small comments.

@fsbraun Thanks, I've done the suggested changes!

File: cms/forms/fields.py [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
 from cms.forms.validators import validate_url_extra
 from cms.forms.widgets import PageSelectWidget, PageSmartLinkWidget
 from cms.models.pagemodel import Page
+from cms.utils.compat import DJANGO_4_2
 
 
 class SuperLazyIterator:
     def __init__(self, func):
+        warnings.warn("SuperLazyIterator is deprecated.",
+                      DeprecationWarning, stacklevel=2)
```
Comment (fsbraun) [inline cms/forms/fields.py:19]: Sorry, I did not see this. 
```suggestion
```

**Comment by protoroto** [comment]
@fsbraun No worries, I was going to ask if you wanted SuperLazyIterator not deprecated too ;)

**Review by fsbraun — APPROVED** [review]

**Comment by protoroto** [comment]
@fsbraun Do you know why the 6 expected checks don't run? It is because their name is changed in `test.yml`? That change was in a commit I cherry-picked from you, but if here we need the "old" names, I'll change them back.

**Comment by fsbraun** [comment]
No, pls. Keep them.

**Comment by fsbraun** [comment]
Thanks! I will change the requirements for the repo tests to pass once this is merged.

**Comment by marksweb** [comment]
> Do you know why the 6 expected checks don't run? It is because their name is changed in `test.yml`?

Yes that's why. The checks get marked as expected, then if their name changes, you end up in this situation but it's not a problem.

**Comment by protoroto** [comment]
@marksweb @fsbraun Ok, should I do something else here, or it's ok like this?

**Comment by marksweb** [comment]
@protoroto no, I was just waiting for all the checks to pass.

**Pull request merged**